### PR TITLE
Add dev tools endpoint for resetting a team's data

### PR DESCRIPTION
**Pull request by benilovj**
This will make it easier to run repeatable black box tests of the app.

I'm not convinced this is the best way to implement it – open to suggestions on alternatives.

### CODE REVIEW

File: app/controllers/dev_controller.rb [hunk 1]
```diff
@@ -16,6 +16,63 @@ def reset
...
+      .joins(:cohort)
+      .where(cohorts: { team: })
+      .distinct
+      .find_each do |patient|
+        parent_ids = patient.parent_ids
+        patient.parents.clear
+        parent_ids.each { |parent_id| Parent.find(parent_id).destroy }
+
+        patient.save!
+        patient.destroy!
```
Comment (thomasleese) [inline app/controllers/dev_controller.rb:68]: I wonder if it's worth setting up the `has_many`s in a way that allows us to do `patient.parents.destroy_all` and then `patient.destroy!`, I think that would read better here, but not a big deal.

Comment (benilovj) [reply app/controllers/dev_controller.rb]: Like this? [200211e](https://github.com/nhsuk/manage-vaccinations-in-schools/pull/1882/commits/200211efeafbeab7b7bef47a7675990c4ede1323)

**Review by tvararu — APPROVED** [review]

File: app/controllers/dev_controller.rb (hunk 1 above)
Comment (thomasleese) [reply app/controllers/dev_controller.rb]: That works, although I was thinking we could do something like:

```rb
# models/patient.rb
has_many :parent_relationships, dependent: :destroy
```

```rb
patients.destroy_all
```

```rb
# Delete childless parents
Parent.joins(:patients).where(patients: { id: nil }).destroy_all
```

Comment (benilovj) [reply app/controllers/dev_controller.rb]: I've left it as is for now just to keep things moving, but not precious about it if anyone changes it afterwards.

**Pull request merged**